### PR TITLE
Begin generating index CRUD operations

### PR DESCRIPTION
**Pull request by Xtansia**
### Description
Begin generating indices.get, shortcutting a few of the more expansive types for property mapping and query types etc. Will make follow up PRs to generate them.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dblock — APPROVED** [review]

File: UPGRADING.md [hunk 1]
```diff
@@ -51,4 +51,13 @@ After:
 - The `name` field, getter and builder method have been renamed to `repository`.
 
 ### CloneSnapshotRequest timeout removal
-- The `timeout` field, getter and builder method have been removed from `CloneSnapshotRequest` as it is not supported by OpenSearch.
\ No newline at end of file
+- The `timeout` field, getter and builder method have been removed from `CloneSnapshotRequest` as it is not supported by OpenSearch.
+
+### indices.Alias
+- The `indexRouting`, `routing` and `searchRouting` properties now accept a `List<String>` instead of a `String`.
```
Comment (dblock) [inline UPGRADING.md:57]: For later, this could be patched to accept `String` as well, right?

File: UPGRADING.md [hunk 2]
```diff
@@ -51,4 +51,13 @@ After:
...
+
+### InlineScript
+- The `lang` property now accepts a `ScriptLanguage` enum instead of a `String`.
```
Comment (dblock) [inline UPGRADING.md:60]: This too could be made backwards compatible with a patch?

File: java-client/src/generated/java/org/opensearch/client/opensearch/_types/ExpandWildcard.java [hunk 3]
```diff
@@ -30,32 +30,33 @@
...
 import org.opensearch.client.json.JsonpDeserializable;
 
+// typedef: _types.ExpandWildcard
+
 @JsonpDeserializable
+@Generated("org.opensearch.client.codegen.CodeGenerator")
 public enum ExpandWildcard implements JsonEnum {
     /**
-     * Match any data stream or index, including hidden ones.
+     * Match any index, including hidden ones.
```
Comment (reta) [inline java-client/src/generated/java/org/opensearch/client/opensearch/_types/ExpandWildcard.java:49]: Could we bring back the previous comment? The note about data streams is important I think

File: java-client/src/generated/java/org/opensearch/client/opensearch/_types/Script.java [hunk 4]
```diff
@@ -165,4 +169,18 @@ private static JsonpDeserializer<Script> buildScriptDeserializer() {
...
     public static final JsonpDeserializer<Script> _DESERIALIZER = JsonpDeserializer.lazy(Script::buildScriptDeserializer);
+
+    public int hashCode() {
+        int result = 17;
+        result = 31 * result + Objects.hashCode(this._kind);
+        result = 31 * result + Objects.hashCode(this._value);
+        return result;
+    }
+
+    public boolean equals(Object o) {
```
Comment (reta) [inline java-client/src/generated/java/org/opensearch/client/opensearch/_types/Script.java:180]:
```suggestion
    @Override
    public boolean equals(Object o) {
```

File: java-client/src/generated/java/org/opensearch/client/opensearch/_types/Script.java [hunk 5]
```diff
@@ -165,4 +169,18 @@ private static JsonpDeserializer<Script> buildScriptDeserializer() {
     }
 
     public static final JsonpDeserializer<Script> _DESERIALIZER = JsonpDeserializer.lazy(Script::buildScriptDeserializer);
+
+    public int hashCode() {
```
Comment (reta) [inline java-client/src/generated/java/org/opensearch/client/opensearch/_types/Script.java:173]:
```suggestion
    @Override
    public int hashCode() {
```

File: java-client/src/generated/java/org/opensearch/client/opensearch/_types/ScriptBase.java [hunk 6]
```diff
@@ -45,82 +52,109 @@
...
     }
 
     // ---------------------------------------------------------------------------------------------
-    protected static <BuilderT extends AbstractBuilder<BuilderT>> void setupScriptBaseDeserializer(ObjectDeserializer<BuilderT> op) {
 
+    protected static <BuilderT extends AbstractBuilder<BuilderT>> void setupScriptBaseDeserializer(ObjectDeserializer<BuilderT> op) {
         op.add(AbstractBuilder::params, JsonpDeserializer.stringMapDeserializer(JsonData._DESERIALIZER), "params");
+    }
 
+    public int hashCode() {
```
Comment (reta) [inline java-client/src/generated/java/org/opensearch/client/opensearch/_types/ScriptBase.java:148]:
```suggestion
    @Override
    public int hashCode() {
```

File: java-client/src/generated/java/org/opensearch/client/opensearch/_types/ScriptBase.java [hunk 7]
```diff
@@ -45,82 +52,109 @@
...
+        int result = 17;
+        result = 31 * result + Objects.hashCode(this.params);
+        return result;
     }
 
+    public boolean equals(Object o) {
```
Comment (reta) [inline java-client/src/generated/java/org/opensearch/client/opensearch/_types/ScriptBase.java:154]:
```suggestion
    @Override
    public boolean equals(Object o) {
```

File: java-client/src/generated/java/org/opensearch/client/opensearch/_types/mapping/RuntimeField.java [hunk 8]
```diff
@@ -0,0 +1,389 @@
...
+import org.opensearch.client.opensearch._types.Script;
+import org.opensearch.client.util.ApiTypeHelper;
+import org.opensearch.client.util.ObjectBuilder;
+import org.opensearch.client.util.ObjectBuilderBase;
+
+// typedef: _types.mapping.RuntimeField
+
+@JsonpDeserializable
+@Generated("org.opensearch.client.codegen.CodeGenerator")
+public class RuntimeField implements PlainJsonSerializable {
```
Comment (reta) [inline java-client/src/generated/java/org/opensearch/client/opensearch/_types/mapping/RuntimeField.java:61]: We don't support RuntimeFields, the equivalent feature is derived fields (see please https://github.com/opensearch-project/OpenSearch/issues/12281)

File: java-client/src/generated/java/org/opensearch/client/opensearch/_types/mapping/SourceFieldMode.java [hunk 9]
```diff
@@ -30,20 +30,36 @@
...
-    public static final ExistsResponse _INSTANCE = new ExistsResponse();
+@JsonpDeserializable
+@Generated("org.opensearch.client.codegen.CodeGenerator")
+public enum SourceFieldMode implements JsonEnum {
+    Disabled("disabled"),
 
-    public static final JsonpDeserializer<ExistsResponse> _DESERIALIZER = JsonpDeserializer.fixedValue(ExistsResponse._INSTANCE);
+    Stored("stored"),
 
+    Synthetic("synthetic");
```
Comment (reta) [inline java-client/src/generated/java/org/opensearch/client/opensearch/_types/mapping/SourceFieldMode.java:52]: I think we don't support `synthetic` either

File: UPGRADING.md (hunk 1 above)
Comment (reta) [reply UPGRADING.md:57]: Why is that? I believe it does not match the server model (`AliasActions`) where this properties are `String`s

File: java-client/src/generated/java/org/opensearch/client/opensearch/indices/Alias.java [hunk 10]
```diff
@@ -42,45 +50,46 @@
...
 @JsonpDeserializable
+@Generated("org.opensearch.client.codegen.CodeGenerator")
 public class Alias implements PlainJsonSerializable {
+
     @Nullable
     private final Query filter;
 
-    @Nullable
-    private final String indexRouting;
+    @Nonnull
```
Comment (reta) [inline java-client/src/generated/java/org/opensearch/client/opensearch/indices/Alias.java:66]: I believe this is not the shape of the API we support (https://www.elastic.co/guide/en/elasticsearch/reference/7.12/indices-aliases.html)

File: UPGRADING.md (hunk 1 above)
Comment (Xtansia) [reply UPGRADING.md]: Have undone this change / tweaked the spec. Was a case of a re-used schema between the query string params which accept a comma-separated list and the body fields which only accept/return strings.

File: java-client/src/generated/java/org/opensearch/client/opensearch/_types/ExpandWildcard.java (hunk 3 above)
Comment (Xtansia) [reply java-client/src/generated/java/org/opensearch/client/opensearch/_types/ExpandWildcard.java:49]: I'm unable to find any reference to expand wildcard in the context of data streams in any OS docs or in the source code. So I'm not sure if those comments are actually relevant to OS or if they're related to the expanded data stream APIs added in ES.

File: java-client/src/generated/java/org/opensearch/client/opensearch/_types/mapping/SourceFieldMode.java (hunk 9 above)
Comment (Xtansia) [reply java-client/src/generated/java/org/opensearch/client/opensearch/_types/mapping/SourceFieldMode.java]: Yeah, looked into it and this enum and the associated `mode` property on `SourceField` don't exist in OS.

Have made a PR removing it from the spec (https://github.com/opensearch-project/opensearch-api-specification/pull/652) and removed it here.

**Comment by Xtansia** [comment]
@dblock @reta Could you please take another look at this? I think I've covered all the comments/concerns.

**Review by reta — APPROVED** [review]

**Review by dblock — APPROVED** [review]

**Pull request merged**